subsys/bluetooth/Kconfig

```diff
@@ -124,14 +124,12 @@ config BT_REMOTE_VERSION
 
 config BT_PHY_UPDATE
 	bool "PHY Update"
-	depends on BT_CTLR_PHY_UPDATE_SUPPORT
```

Comment:
Should this be:
`depends on BT_CTLR_PHY_UPDATE_SUPPORT if BT_CTLR`?
same for the other one.

Reply:
Possibly. That would require everyone that selects the `BT_CTLR` option to use the `BT_CTLR_<feature>_SUPPORT` options as well. If this is what we want I can update to that.

Reply:
Can't put an `if` on a `depends on`, so would need to make it something like `depends on !(BT_CTLR && !BT_CTLR_PHY_UPDATE_SUPPORT)`.

Don't know anything about these symbols. Just a drive-by note.

Reply:
Thanks @ulfalizer 

@carlescufi I don't see the need to be so strict with the option here, the host will use read supported features in addition.

Reply:
Well at the very least we should then do:

```
select BT_CTLR_PHY_UPDATE_SUPPORT  if BT_CTLR
```

or something similar. In general for combined builds we want users to be able to use a single switch to enable a particular feature, not having to use two different ones. Can you compare how we do it with other features?

Reply:
>depends on BT_CTLR_PHY_UPDATE_SUPPORT if BT_CTLR

@joerchan @carlescufi how would this work with host and controller on different images? Pls, explain if possible

Reply:
@ioannisg The host should check support at runtime using HCI command to read
attached controller features and commands supported. In a host-only image this feature should not have to depend on any controller configuration at build time.

If you mean host and controller image built to be feature compatible then that is not the problem I'm trying to solve here.

@carlescufi Your suggested `select BT_CTLR_PHY_UPDATE_SUPPORT  if BT_CTLR` is not how that is supposed to be used. The different BT_CTLR_VENDORS_* selects the features it supports. The suggestion by Ulf can work `depends on !(BT_CTLR && !BT_CTLR_PHY_UPDATE_SUPPORT)` but I'm still not convinced it is the best idea.

Reply:
That was my point, actually, @joerchan ; it does not seem to me that `depends on BT_CTLR_PHY_UPDATE_SUPPORT if BT_CTLR` is the right thing to do, because the image with the BT host might not have BT_CTLR set, if controller is on another image (i wanted you to verify that my thinking is correct).

Agree also with `BT_CTLR_xxxx_SUPPORT `symbols; these indicated vendor capabilities and should be used in the controller side, right?

Reply:
@ioannisg As Ulf explained `depends on BT_CTLR_PHY_UPDATE_SUPPORT if BT_CTLR` does not work. The other suggestion is `depends on !(BT_CTLR && !BT_CTLR_PHY_UPDATE_SUPPORT)`.
Your thinking is correct, in a host-only build BT_CTLR is not set so `depends on` expression must evaluate to true in that case.
I'm still in favour of not having any depends on for these symbols however. So now I'm just waiting for @carlescufi response to this thread.

> Agree also with BT_CTLR_xxxx_SUPPORT symbols; these indicated vendor capabilities and should be used in the controller side, right?

Yes that is right.